### PR TITLE
Add debug variants of all containers that include busybox.

### PR DESCRIPTION
**Pull request by mattmoor**
Switch from publishing things through `cloudbuild.yaml` explicitly to creating a bundle and publishing all of the images within.

### CODE REVIEW

**Comment by mattmoor** [comment]
As context, I'd like to enable using the debug container for debug builds of `lang_image` rules.

**Comment by mattmoor** [comment]
No clue why travis failed, I'll kick it.

**Comment by mattmoor** [comment]
@dlorenc Do you have any idea why this might be failing?

**Comment by mattmoor** [comment]
[This PR](https://github.com/bazelbuild/rules_docker/pull/184) shows how I plan to make use of these.

**Comment by r2d4** [comment]
Looks like buildifier errors, not sure why the error message didnt show up on travis though.

**Comment by mattmoor** [comment]
Alright, `buildifier` issues fixed.  It is taking the first `exit 1` in `buildifier.sh`, which is why there is no output.

**Comment by mattmoor** [comment]
@r2d4 This should be RFAL if you want to do the honors?

File: cc/BUILD [hunk 1]
```diff
@@ -4,11 +4,14 @@ load("@io_bazel_rules_docker//docker:docker.bzl", "docker_build")
 load("@package_bundle//file:packages.bzl", "packages")
 
 # An intermediate image for Java and other "mostly statically" compiled languages
-docker_build(
-    name = "cc",
-    base = "//base:base",
+[docker_build(
+    name = "cc" if not mode else mode[1:],
```
Comment (r2d4) [inline cc/BUILD]: Does it make more sense to iterate the other way?
```
docker_build(
  name = image + ":debug"
  base = "//" + image
  ...
  directory = "/",
  entrypoint = ["/busybox/sh"],
  env = {"PATH": "/busybox"},
  tars = ["//busybox:busybox.tar"],
...
for image in [
   "cc",
   "java",
   "python",
   ...
```

Comment (mattmoor) [reply cc/BUILD]: Name can't have a `:` in it.  It also can't have a `/` so things start getting awkward for `java/jetty:debug`.  I also like the symmetry of having the tags match the BUILD labels, but I'm certainly not married to it given another suggestion.

Comment (mattmoor) [reply cc/BUILD]: Things are further complicated by the lack of symmetry for targets like `java:java8` and `python2.7:python27` :(

Comment (r2d4) [reply cc/BUILD]: That make sense.  Just wondering if there was a way we could do it without writing the code for each image.  LGTM

Comment (mattmoor) [reply cc/BUILD]: I was debating writing a little skylark macro for it, but the quirk in my second comment made that hard enough to stop me.  If it were just `//foo` and `//foo:debug` everywhere I probably would have.  I don't recall the history of why we let these diverge...

**Comment by mattmoor** [comment]
Ok, I'm gonna merge to unblock https://github.com/bazelbuild/rules_docker/pull/184

**Pull request merged**